rules/coding-style/src/Rector/Variable/UnderscoreToPascalCasePropertyNameRector.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\Variable;
+
+use Nette\Utils\Strings;
+use PhpParser\Node;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\StaticPropertyFetch;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Stmt\PropertyProperty;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\Core\Util\StaticRectorStrings;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+/**
+ * @see \Rector\CodingStyle\Tests\Rector\Variable\UnderscoreToPascalCaseVariableAndPropertyNameRector\UnderscoreToPascalCaseVariableAndPropertyNameRectorTest
+ */
+final class UnderscoreToPascalCasePropertyNameRector extends AbstractRector
+{
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Change under_score names to pascalCase', [
+            new CodeSample(
+                <<<'PHP'
+final class SomeClass
+{
+    public $property_name;
+
+    public function run($a)
+    {
+        $this->property_name = 5;
+    }
+}
+PHP
+,
+                <<<'PHP'
+final class SomeClass
+{
+    public $propertyName;
+
+    public function run($a)
+    {
+        $this->propertyName = 5;
+    }
+}
+PHP
+
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [PropertyProperty::class, PropertyFetch::class, StaticPropertyFetch::class];
+    }
+
+    /**
+     * @param PropertyProperty|PropertyFetch|StaticPropertyFetch $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $nodeName = $this->getName($node);
+        if ($nodeName === null) {
+            return null;
+        }
+
+        /** @var string $class */
+        $class = $node->getAttribute(AttributeKey::CLASS_NAME);
+        // properties are accessed via magic, nothing we can do
+        if (method_exists($class, '__set') || method_exists($class, '__get')) {
+            return null;
+        }
+
+        // @see https://regex101.com/r/dXc89K/1
+        // matches if underscore is present but excluding first underscore
+        if (! Strings::match($nodeName, '#(?<!^)\_#')) {
+            return null;
+        }
+
+        $camelCaseName = $this->createPascalName($nodeName, $node);
+
+        $node->name = new Identifier($camelCaseName);
+
+        return $node;
+    }
+
+    /**
+     * @param PropertyProperty|PropertyFetch|StaticPropertyFetch $node
+     */
+    private function createPascalName(string $nodeName, Node $node): string
+    {
+        $camelCaseName = StaticRectorStrings::underscoreToPascalCase($nodeName);
```

Comment:
@TomasVotruba, don't you think that `StaticRectorStrings::underscoreToPascalCase` shouldn't uppercase first letter in case there is _ before it? Or it might be a good idea to create StaticRectorVariable::underscoreToPascalCase which will care of it...

Reply:
By coding standar, property and variable names should always start with lower letter.

Reply:
It's probably incorrectly user here. Could you add test fixture for it and fix it?

Reply:
Not sure what do you mean. 
Here is how it works at the moment
```diff
-    public $some_property;
+    public $someProperty;

-    public static $some_static_property;
+    public static $someStaticProperty;

-    public $_first_and_multiple_underscore_property;
+    public $_firstAndMultipleUnderscoreProperty;

-    public $_underscore;
+    public $_underscore;
```
Is it correct behavior?
This behavior is covered by test fixtures.

Reply:
I mean like this:

```diff
-    public $_first_and_multiple_underscore_property;
+    public $firstAndMultipleUnderscoreProperty;

-    public $_underscore;
+    public $underscore;
```

Reply:
Just check the PR :)

Reply:
Just checked, it's missing there.

Reply:
In original rule properties and variable names with _ in the beginning, are skipped. That is why I left the original behavior and made these rule skip properties which are accessed using magic calls. So for now as far as I understand the rule should skip only "magic accesses'?

Reply:
Also... by removing underscore in the beginning there might be a potential problem. In case, in the code (for some crazy reason) there are two variables/properties one `$_some_value` and another `$some_value`  the rule will rename to the same name `$someValue`